### PR TITLE
DOC-1812  - replaced existing 'Persisting Hierarchical Data' section with suggested working example at GitHub link in ticket

### PR DESCRIPTION
**Pull request by paulmoloneyr3**
See:

https://github.com/corda/corda/issues/5254#issuecomment-794497804

### CODE REVIEW

File: content/en/platform/corda/4.0/enterprise/api-persistence.md [hunk 1]
```diff
@@ -226,219 +226,62 @@ e.g. for Oracle database, prior to version 12.2 the maximum length of table/colu
 
 ## Persisting Hierarchical Data
 
-You may wish to persist hierarchical relationships within states using multiple database tables
+You may wish to persist hierarchical relationships within state data using multiple database tables.
```
Comment (nadinequinr3) [inline content/en/platform/corda/4.0/enterprise/api-persistence.md:229]:
```suggestion
You may want to persist hierarchical relationships within state data using multiple database tables. 
```

File: content/en/platform/corda/4.0/enterprise/api-persistence.md [hunk 2]
```diff
@@ -226,219 +226,62 @@ e.g. for Oracle database, prior to version 12.2 the maximum length of table/colu
 
 ## Persisting Hierarchical Data
 
-You may wish to persist hierarchical relationships within states using multiple database tables
+You may wish to persist hierarchical relationships within state data using multiple database tables. 
+In order to facilitate this, you must implement all queries making use of hierarchical relations as native SQL.
```
Comment (nadinequinr3) [inline content/en/platform/corda/4.0/enterprise/api-persistence.md:230]:
```suggestion
To facilitate this, you must implement all queries making use of hierarchical relations as native SQL. 
```

File: content/en/platform/corda/4.1/enterprise/api-persistence.md [hunk 3]
```diff
@@ -285,219 +285,62 @@ e.g. for Oracle database, prior to version 12.2 the maximum length of table/colu
...
+In order to facilitate this, you must implement all queries making use of hierarchical relations as native SQL. 
 
-You may wish to persist hierarchical relationships within state data using multiple database tables. In order to facillitate this, multiple `PersistentState`
-subclasses may be implemented. The relationship between these classes is defined using JPA annotations. It is important to note that the `MappedSchema`
-constructor requires a list of *all* of these subclasses.
+Example schemas implementing hierarchical relationships have been implemented below. 
 
-An example Schema implementing hierarchical relationships with JPA annotations has been implemented below. This Schema will cause `parent_data` and `child_data` tables to be
-created.
+For one-to-many scenarios, using the following:
```
Comment (yarivus) [inline content/en/platform/corda/4.1/enterprise/api-persistence.md:293]: Isn't this line repeated ? see below (L293, L315)

**Review by yarivus — CHANGES_REQUESTED** [review]

File: content/en/platform/corda/4.0/enterprise/api-persistence.md [hunk 4]
```diff
@@ -226,219 +226,62 @@ e.g. for Oracle database, prior to version 12.2 the maximum length of table/colu
...
+In order to facilitate this, you must implement all queries making use of hierarchical relations as native SQL. 
 
-You may wish to persist hierarchical relationships within state data using multiple database tables. In order to facillitate this, multiple `PersistentState`
-subclasses may be implemented. The relationship between these classes is defined using JPA annotations. It is important to note that the `MappedSchema`
-constructor requires a list of *all* of these subclasses.
+Example schemas implementing hierarchical relationships have been implemented below. 
 
-An example Schema implementing hierarchical relationships with JPA annotations has been implemented below. This Schema will cause `parent_data` and `child_data` tables to be
-created.
+For one-to-many scenarios, using the following:
```
Comment (nadinequinr3) [inline content/en/platform/corda/4.0/enterprise/api-persistence.md:234]: use?

File: content/en/platform/corda/4.1/enterprise/api-persistence.md (hunk 3 above)
Comment (paulmoloneyr3) [reply content/en/platform/corda/4.1/enterprise/api-persistence.md:293]: Gosh yes, meant to be one-to-one. Let me refix those.

**Review by nadinequinr3 — CHANGES_REQUESTED** [review]
I haven't repeated my comments for every instance. I assume the same apply to all.
Also, did you verify with @colin-hillis how many versions back we have to go?

File: content/en/platform/corda/4.1/enterprise/api-persistence.md (hunk 3 above)
Comment (paulmoloneyr3) [reply content/en/platform/corda/4.1/enterprise/api-persistence.md]: Resolved.

File: content/en/platform/corda/4.0/enterprise/api-persistence.md (hunk 4 above)
Comment (paulmoloneyr3) [reply content/en/platform/corda/4.0/enterprise/api-persistence.md]: Fixed now.

File: content/en/platform/corda/4.0/enterprise/api-persistence.md (hunk 2 above)
Comment (paulmoloneyr3) [reply content/en/platform/corda/4.0/enterprise/api-persistence.md]: Fixed.

File: content/en/platform/corda/4.0/enterprise/api-persistence.md (hunk 1 above)
Comment (paulmoloneyr3) [reply content/en/platform/corda/4.0/enterprise/api-persistence.md]: Fixed above.

**Review by AnnaK-R3 — APPROVED** [review]

**Review by nadinequinr3 — APPROVED** [review]

**Pull request merged**